### PR TITLE
bloop: 1.4.9 -> 1.4.11

### PR DESCRIPTION
**Pull request by kubukoz**
###### Motivation for this change

Update Bloop

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Tomahna — APPROVED** [review]

**Review by JesusMtnez — APPROVED** [review]

**Comment by Tomahna** [comment]
@kubukoz Looks good, thanks for the PR
To make things a bit faster, you can opt in for marvin https://github.com/apps/marvin-mk2

File: pkgs/development/tools/build-managers/bloop/default.nix [hunk 1]
```diff
@@ -10,34 +10,34 @@
...
   bloop-coursier = stdenv.mkDerivation rec {
     name = "${pname}-coursier-${version}";
 
     platform = if stdenv.isLinux && stdenv.isx86_64 then "x86_64-pc-linux"
-               else if stdenv.isDarwin && stdenv.isx86_64 then "x86_64-apple-darwin"
-               else throw "unsupported platform";
+    else if stdenv.isDarwin && stdenv.isx86_64 then "x86_64-apple-darwin"
+    else throw "unsupported platform";
 
     phases = [ "installPhase" ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/build-managers/bloop/default.nix:42]:
```suggestion
```
Overwriting phases in packages is strictly forbidden.

Comment (kubukoz) [reply pkgs/development/tools/build-managers/bloop/default.nix]: Fixed!

**Comment by kubukoz** [comment]
Also added pre/post-install hook calls like in #142850.

**Comment by Tomahna** [comment]
Tested it on nixos and macos (with home-manager)

On nixos, everything works fine.

On mac I get an error when building:
```
post-installation fixup
strip is /nix/store/5j78r51lhl6pwfjxd2cz25p40xcnnywn-cctools-binutils-darwin-949.0.1/bin/strip
patching script interpreter paths in /nix/store/cd9p7062v4giqjj78828wvx3dmm89f04-bloop-coursier-1.4.11
/nix/store/cd9p7062v4giqjj78828wvx3dmm89f04-bloop-coursier-1.4.11/bloop: interpreter directive changed from "#!/usr/bin/env sh" to "/nix/store/vm564n0g8k6pkiwqb31x1dw1y02h2mak-bash-5.1-p8/bin/sh"
error: path '/nix/store/cd9p7062v4giqjj78828wvx3dmm89f04-bloop-coursier-1.4.11' is not valid
```

I'm unclear if it is specific to my setup or not, I'll try to investigate more tomorrow.

**Comment by kubukoz** [comment]
Are you on nix 2.4? I'm pretty sure this is a bug in nix itself. Works fine on 2.3.15 AFAIK

**Comment by Tomahna** [comment]
Running `nix upgrade-nix` "upgraded" nix from 2.4 to 2.3.16 and I have still the same issue.

**Comment by kubukoz** [comment]
If you're on multi user mode, can you double check if the daemon also runs the same version? (I had some trouble with that when 2.4 came)

Anyway I'll try building this from scratch again :)

**Comment by Tomahna** [comment]
Indeed, restarting the daemon did solve the issue.

**Review by Tomahna — APPROVED** [review]

**Comment by kubukoz** [comment]
Updated to avoid relying on the provided wrapper. This allows us to stop doing weird stuff with sed, and also makes the intermediate derivation valid on 2.4, as it no longer refers to itself.

Please review :)

**Review by Tomahna — APPROVED** [review]
Tested on Nixos and Mac

**Comment by kubukoz** [comment]
@SuperSandro2000 can you merge this please?

**Comment by jonringer** [comment]
trusting @Tomahna 's review :)

**Comment by kubukoz** [comment]
Be careful, we know each other :joy:

**Comment by jonringer** [comment]
All good then :)

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 144673 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>1 package built:</summary>  <ul>    <li>bloop</li>  </ul></details>

**Pull request merged**